esp_encrypted_img/include/esp_encrypted_img.h

```diff
@@ -181,6 +181,17 @@ esp_err_t esp_encrypted_img_decrypt_abort(esp_decrypt_handle_t ctx);
 */
 uint16_t esp_encrypted_img_get_header_size(void);
 
+/**
+ * @brief  Get the public key corresponding to the private key.
+ *         For RSA, the public key is in DER format and corresponds to the private key passed with `esp_encrypted_img_decrypt_start()`.
+ *         For ECIES, the public key is in uncompressed format and is derived from the HMAC key ID passed with `esp_encrypted_img_decrypt_start()`.
+ * 
+ * @param ctx   esp_decrypt_handle_t handle
+ * @param pub_key_len   Pointer to store the length of the public key
+ * @return uint8_t*   Pointer to the public key
+ */
+uint8_t *esp_encrypted_img_get_public_key(esp_decrypt_handle_t ctx, size_t *pub_key_len);
```

Comment:
- It will be application layer responsibility to free the pointer after usage? If yes, please document.
- Suggest changing API to return `esp_err_t` and then signature to below
- Also suggest using export instead of get in the API name

```
esp_err_t esp_encrypted_img_export_public_key(esp_decrypt_handle_t handle, uint8_t **pub_key, size_t *pub_key_len)
```